mypy/messages.py

```diff
@@ -2394,23 +2395,29 @@ def [T <: int] f(self, x: int, y: T) -> None
         s += format_type_bare(tp.arg_types[i])
         if tp.arg_kinds[i].is_optional():
             s += " = ..."
+        if (
+            not slash
+            and tp.arg_kinds[i].is_positional()
+            and name is None
+            and (
+                i == len(tp.arg_types) - 1
+                or (tp.arg_names[i + 1] is not None or not tp.arg_kinds[i + 1].is_positional())
+            )
+        ):
+            s += ", /"
+            slash = True
 
     # If we got a "special arg" (i.e: self, cls, etc...), prepend it to the arg list
     if (
         isinstance(tp.definition, FuncDef)
         and tp.definition.name is not None
         and hasattr(tp.definition, "arguments")
     ):
-        definition_args = [arg.variable.name for arg in tp.definition.arguments]
-        if (
-            definition_args
-            and tp.arg_names != definition_args
-            and len(definition_args) > 0
-            and definition_args[0]
-        ):
+        definition_arg_names = [arg.variable.name for arg in tp.definition.arguments]
+        if len(definition_arg_names) > len(tp.arg_names) and definition_arg_names[0]:
```

Comment:
Btw is it true that `len(definition_arg_names) == len(tp.arg_names) or len(definition_arg_names) == len(tp.arg_names) + 1`? i.e. is there ever some case with multiple special args?

Reply:
I don't think there are any such cases.